sonic-utilities-tests/config_mgmt_test.py

```diff
@@ -0,0 +1,475 @@
+        self.dpb_port8_2x50G_1x100G(iConfig)
+        # Ethernet8: move from 2x50G-->1x100G with force, where deps exists
+        self.dpb_port8_2x50G_1x100G_f(iConfig)
+        # Ethernet8: move from 1x100G-->4x25G without force, no deps
+        self.dpb_port8_1x100G_4x25G(iConfig)
+        # Ethernet8: move from 4x25G-->1x100G with force, no deps
+        self.dpb_port8_4x25G_1x100G_f(iConfig)
+        # Ethernet8: move from 1x100G-->1x50G(2)+2x25G(2) with -f -l,
+        self.dpb_port8_1x100G_1x50G_2x25G_f_l(iConfig)
+        # Ethernet4: breakout from 4x25G to 2x50G with -f -l
+        self.dpb_port4_4x25G_2x50G_f_l(iConfig)
+        return
+
+    def tearDown(self):
+        try:
+            os.remove(config_mgmt.CONFIG_DB_JSON_FILE)
+            os.remove(config_mgmt.DEFAULT_CONFIG_DB_JSON_FILE)
+        except Exception as e:
+            pass
+        return
+
+    ########### HELPER FUNCS #####################################
+    def writeJson(self, d, file):
+        with open(file, 'w') as f:
+            dump(d, f, indent=4)
+        return
+    # config_mgmt.ConfigMgmtDPB class instance with mocked functions. Not using
+    # pytest fixture, because it is used in non test funcs.
+    def config_mgmt_dpb(self, iConfig):
+        # create object
+        self.writeJson(iConfig, config_mgmt.CONFIG_DB_JSON_FILE)
+        cmdpb = config_mgmt.ConfigMgmtDPB(source=config_mgmt.CONFIG_DB_JSON_FILE)
+        # mock funcs
+        cmdpb.writeConfigDB = MagicMock(return_value=True)
+        cmdpb._verifyAsicDB = MagicMock(return_value=True)
+        return cmdpb
+
+    # Generate port to deleted, added and lanes and speed setting based on
+    # current and new mode.
+    def generate_args(self, portIdx, laneIdx, curMode, newMode):
+        # default params
+        pre = "Ethernet"
+        laneMap = {"4x25G": [1,1,1,1], "2x50G": [2,2], "1x100G":[4], \
+            "1x50G(2)+2x25G(2)":[2,1,1], "2x25G(2)+1x50G(2)":[1,1,2]}
+        laneSpeed = 25000
+        # generate dPorts
+        l = list(laneMap[curMode]); l.insert(0, 0); id = portIdx; dPorts = list()
+        for i in l[:-1]:
+            id = id + i
+            portName = portName = "{}{}".format(pre, id)
+            dPorts.append(portName)
+        # generate aPorts
+        l = list(laneMap[newMode]); l.insert(0, 0); id = portIdx; aPorts = list()
+        for i in l[:-1]:
+            id = id + i
+            portName = portName = "{}{}".format(pre, id)
+            aPorts.append(portName)
+        # generate pJson
+        l = laneMap[newMode]; pJson = {"PORT": {}}; li = laneIdx; pi = 0
+        for i in l:
+            speed = laneSpeed*i
+            lanes = [str(li+j) for j in range(i)]; lanes = ','.join(lanes)
+            pJson['PORT'][aPorts[pi]] = {"speed": str(speed), "lanes": str(lanes)}
+            li = li+i; pi = pi + 1
+        return dPorts, aPorts ,pJson
+
+    # update the config to emulate continous breakingout a single port
```

Comment:
This function simulates update in configDB.

-- Assume conf is the current config in ConfigDB, 
-- then uconf is written in ConfigDB, [in the format excepted by mod_config()]
-- then configDB will look like as conf dict() at end of this function. i.e. conf dict will be updated as conf+uconf